lib/goth/config.ex

```diff
@@ -79,12 +79,28 @@ defmodule Goth.Config do
   end
 
   defp determine_project_id(config) do
-    Application.get_env(:goth, :project_id) ||
-      System.get_env("GOOGLE_CLOUD_PROJECT") ||
-      System.get_env("GCLOUD_PROJECT") ||
-      System.get_env("DEVSHELL_PROJECT_ID") ||
-      config["project_id"] ||
-      Client.retrieve_metadata_project()
+    case Application.get_env(:goth, :project_id) || System.get_env("GOOGLE_CLOUD_PROJECT") ||
+           System.get_env("GCLOUD_PROJECT") || System.get_env("DEVSHELL_PROJECT_ID") ||
+           config["project_id"] do
+      nil ->
+        try do
+          Client.retrieve_metadata_project()
+        rescue
+          e in HTTPoison.Error ->
+            case e do
+              %HTTPoison.Error{reason: :nxdomain} ->
+                raise " Failed to retrieve project data from GCE internal metadata service.
```

Comment:
So what error is then thrown if `APPLICATION_DEFAULT_CREDENTIALS` aren't set locally?

Reply:
It seems like a couple things should lead to an error. @peburrows please correct me if I'm wrong, but I think there should be an error if either
1. your Google application credentials aren't specified and can't be fetched from the metadata server
2. your project ID isn't specified, and can't be inferred from your credentials.

So maybe two errors, one for each of these events, would be more appropriate.

Reply:
Sorry for the slow response, @p-buse, but I think this particular change actually covers both those cases.

Reply:
And I'm fine with this single error — the message is clear enough to point someone in the right direction.